--- gst-plugins-good-0.10.17/configure.ac~	2010-01-29 22:06:45.798536762 +0100
+++ gst-plugins-good-0.10.17/configure.ac	2010-01-29 22:07:06.241451973 +0100
@@ -750,7 +750,7 @@
 dnl *** libpng ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_LIBPNG, true)
 AG_GST_CHECK_FEATURE(LIBPNG, [Portable Network Graphics library], png, [
-  AG_GST_PKG_CHECK_MODULES(LIBPNG, libpng12)
+  AG_GST_PKG_CHECK_MODULES(LIBPNG, libpng)
 ])
 
 dnl *** pulseaudio ***
